sigv4.go

```diff
@@ -135,18 +147,37 @@ func (rt *sigV4RoundTripper) RoundTrip(req *http.Request) (*http.Response, error
 	for _, header := range sigv4HeaderDenylist {
 		signReq.Header.Del(header)
 	}
+	creds, err := rt.creds.Retrieve(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving credentials: %w", err)
+	}
 
-	headers, err := rt.signer.Sign(signReq, seeker, "aps", rt.region, time.Now().UTC())
+	hash := sha256.Sum256(buf.Bytes())
+	strHash := hex.EncodeToString(hash[:])
+	err = rt.signer.SignHTTP(
+		ctx,
+		creds,
+		signReq,
+		strHash,
+		rt.serviceName,
+		rt.region,
+		time.Now().UTC(),
+	)
 	if err != nil {
 		return nil, fmt.Errorf("failed to sign request: %w", err)
 	}
 
 	// Copy over signed headers. Authorization header is not returned by
 	// rt.signer.Sign and needs to be copied separately.
-	for k, v := range headers {
+	for k, v := range signReq.Header {
 		req.Header[textproto.CanonicalMIMEHeaderKey(k)] = v
 	}
 	req.Header.Set("Authorization", signReq.Header.Get("Authorization"))
```

Comment:
you don't need to do this anymore as signReq is modified in place. after signing signReq, you can just do:

```
return rt.next.RoundTrip(signReq)
```

Reply:
I'm not sure about this, the thing is we want to send the `uber-trace-id`? If we don't want to send it if it's "ignored" header then it's ok to remove it. Unit tests show that the header should be in the new req

Reply:
I saw what you did and it looks good. Thanks!